test/test_um2netcdf.py

```diff
@@ -651,6 +662,379 @@ def test_fix_units_do_nothing_no_um_units(ua_plev_cube):
+        points=lat_v_points,
+        standard_name=um2nc.LATITUDE
+    )
+    return lat_v_eg_coord
+
+
+@pytest.fixture
+def lon_u_eg():
+    # iris DimCoord imitating the real
+    # lon_v grid from CM2 (which uses the End Game grid).
+    lon_u_points = np.arange(0, 360, D_LON_N96, dtype="float32")
+
+    lon_u_eg_coord = iris.coords.DimCoord(
+        points=lon_u_points,
+        standard_name=um2nc.LONGITUDE
+    )
+    return lon_u_eg_coord
+
+
+@pytest.fixture
+def lat_standard_nd():
+    # iris DimCoord imitating the standard latitude
+    # grid from ESM1.5 (which uses the New Dynamics grid).
+    lat_points_middle = np.arange(-88.75, 89., D_LAT_N96)
+    lat_points = np.concatenate(([-90],
+                                 lat_points_middle,
+                                 [90.]
+                                 ), dtype="float32")
+
+    lat_standard_nd_coord = iris.coords.DimCoord(
+        points=lat_points,
+        standard_name=um2nc.LATITUDE
+    )
+    return lat_standard_nd_coord
+
+
+@pytest.fixture
+def lon_standard_nd():
+    # iris DimCoord imitating the standard longitude
+    # grid from ESM1.5 (which uses the New Dynamics grid).
+    lon_points = np.arange(0, 360, D_LON_N96, dtype="float32")
+
+    lon_standard_nd_coord = iris.coords.DimCoord(
+        points=lon_points,
+        standard_name=um2nc.LONGITUDE
+    )
+    return lon_standard_nd_coord
+
+
+@pytest.fixture
+def lat_standard_eg():
+    # iris DimCoord imitating the standard latitude
+    # grid from CM2 (which uses the End Game grid).
+    lat_points = np.arange(-90 + 0.5*D_LAT_N96, 90.,
+                           D_LAT_N96, dtype="float32")
+
+    lat_standard_eg_coord = iris.coords.DimCoord(
+        points=lat_points,
+        standard_name=um2nc.LATITUDE
+    )
+    return lat_standard_eg_coord
+
+
+@pytest.fixture
+def lon_standard_eg():
+    # iris DimCoord imitating the standard longitude
+    # grid from CM2 (which uses the End Game grid).
+    lon_points = np.arange(0.5*D_LON_N96, 360, D_LON_N96, dtype="float32")
+    lon_standard_eg_coord = iris.coords.DimCoord(
+        points=lon_points,
+        standard_name=um2nc.LONGITUDE
+    )
+    return lon_standard_eg_coord
+
+
+class DummyCubeWithCoords(DummyCube):
+    # DummyCube with coordinates, which can be filled with
+    # iris.DimCoord objects for testing.
+    def __init__(self, dummy_cube, coords=[]):
+        super().__init__(dummy_cube.item_code,
+                         dummy_cube.var_name,
+                         dummy_cube.attributes,
+                         dummy_cube.units)
+
+        # Set up coordinate dictionary to hold each dummy coordinate, with keys
+        # given by the coordinate names. This ensures that the key used to
+        # access a coordinate via the coord method matches the
+        # coordinate's name.
+        self.coordinate_dict = {}
+        for coord in coords:
+            self.coordinate_dict[coord.name()] = coord
```

Comment:
Is this simpler as a dict comprehension?

Reply:
Yep! Added in in latest commits